pyro/contrib/forecast/evaluate.py

```diff
@@ -0,0 +1,163 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+import torch
+
+import pyro
+from pyro.ops.stats import crps_empirical
+
+from .forecaster import Forecaster
+
+logger = logging.getLogger(__name__)
+
+
+@torch.no_grad()
+def eval_mae(pred, truth):
+    """
+    Evaluate mean absolute error, using sample median as point estimate.
+
+    :param torch.Tensor pred: Forecasted samples.
+    :param torch.Tensor truth: Ground truth.
+    :rtype: float
+    """
+    pred = pred.median(0).values
```

Comment:
Should we also support MAE and RMSE for the full set of predictions (not just the median/mean), or do you think CRPS is better reserved for that?

Reply:
No, MAE and RMSE don't make sense for monte carlo samples. CRPS is a good metric for monte carlo samples.

Reply:
In particular, mean can be seen as a minimizer of RMSE, median as a minimizer of MAE, and correctly-distributed samples as a minimizer of CRPS. Each pair is tightly coupled.

Reply:
Makes sense.